src/opera/compare/comparisons.py

```diff
@@ -0,0 +1,72 @@
+from .diff import Diff
+
+
+class Comparison:
+    def __init__(self, compare_func):
+        self.item_compare_func = compare_func
+
+    def compare(self, item1, item2, context):
+        return self.item_compare_func(item1, item2, context)
+
+
+class ListComparison(Comparison):
+    def __init__(self, compare_func, id_func):
+        super().__init__(compare_func)
+        self.item_id_func = id_func
+
+    def compare(self, list1, list2, context):
+        diff = Diff()
+        for item1 in list1:
+            item2 = next((i for i in list2 if (self.item_id_func(item1) ==
+                                               self.item_id_func(i))), None)
+            if not item2:
+                diff.deleted.append(self.item_id_func(item1))
+            else:
+                equal, change = self.item_compare_func(item1, item2, context)
+                if not equal:
+                    diff.changed[self.item_id_func(item1)] = change
+        for item2 in list2:
+            item1 = next((i for i in list1 if (self.item_id_func(item2) ==
+                                               self.item_id_func(i))), None)
+            if not item1:
+                diff.added.append(self.item_id_func(item2))
+
+        return diff.equal(), diff
+
+
+class MapComparison(Comparison):
+    def __init__(self, compare_func, id_func=None):
+        super().__init__(compare_func)
+        self.item_id_func = id_func
+
+    def compare(self, dict1, dict2, context):
+        if dict1 is None:
+            dict1 = {}
+        if dict2 is None:
+            dict2 = {}
+        # special handling for capability properties and attributes
+        if isinstance(dict1, tuple):
+            dict1 = dict1[0]
+        if isinstance(dict2, tuple):
+            dict2 = dict2[0]
```

Comment:
Why is this special handling needed? Can we update opera parser so that this extra handling will not be necessary?

Reply:
While I agree that should be fixed at some point (if there is no reason for this discrepancy) I do not think it should be done as a part of this PR, but rather in a separate issue.

Reply:
Ok, then it'd be great that you open a new issue just to mention that this is something to be fixed.

Reply:
Will do